clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -26,82 +33,121 @@ class NegativeControlWasPositive(AnalysisServiceException):
     pass
 
 
+class FailedControl(AnalysisServiceException):
+    pass
+
+
 class RTPCRAnalysisService(object):
+    """
+    Create a new RTPCRAnalysis Service:
+        RTPCRAnalysisService(covid_reporter_key="FAM",
+                             internal_control_reporter_key="HEX")
+    """
 
-    HEX_THRESHOLD = 32
-    FAM_THRESHOLD = 38
+    INTERNAL_CONTROL_THRESHOLD = 32
+    COVID_CONTROL_THRESHOLD = 38
 
-    def __init__(self):
-        pass
+    def __init__(self, covid_reporter_key, internal_control_reporter_key):
+        self._covid_reporter_key = covid_reporter_key
+        self._internal_control_reporter = internal_control_reporter_key
 
-# Understanding of the analysis criterias
-# FAM = COVID, HEX = Human internal control
-# Positive if CT(FAM) = <=38
-# Failed  if CT(FAM)= >38
-# Negative if CT(FAM) = 0 and CT(HEX) = <=32
-# Retest if CT(FAM) = 0 and CT(HEX) = >32
+    # These are notes on what Maike said about the analysis
+    # Understanding of the analysis criterias
+    # FAM = COVID, HEX = Human internal control
+    # Positive if CT(FAM) = <=38
+    # Failed  if CT(FAM)= >38
+    # Negative if CT(FAM) = 0 and CT(HEX) = <=32
+    # Failed if CT(FAM) = 0 and CT(HEX) = >32
 
-# TODO Should we issue a re-test status here
-# TODO Later we might have to account for giving two trails over 38 as positive.
+    # TODO Later we might have to account for giving two trails over 38 as positive.
 
     def _analyze_sample(self, sample):
-        fam_ct = sample["FAM-CT"]
-        hex_ct = sample["HEX-CT"]
-        if fam_ct > self.FAM_THRESHOLD:
-            return COVID_RESPONSE_FAILED
-        elif fam_ct == 0 and hex_ct <= self.HEX_THRESHOLD:
+        covid_ct = sample["{}-CT".format(self._covid_reporter_key)]
+        internal_control_ct = sample["{}-CT".format(
+            self._internal_control_reporter)]
+        if covid_ct > self.COVID_CONTROL_THRESHOLD:
```

Comment:
This series of ifs I would suggest that we go through together tomorrow, comparing with the pseudocode.

Reply:
Yes. I think that would be a very good idea!